tools/downloader/src/open_model_zoo/model_tools/_configuration.py

```diff
@@ -361,15 +362,42 @@ def deserialize(cls, model, name, subdirectory):
 
             return cls(name, subdirectory, files, postprocessing, mo_args, framework,
                 description, license_url, precisions, quantization_output_precisions,
-                task_type, conversion_to_onnx_args)
+                task_type, conversion_to_onnx_args, composite_model_name)
+
+def check_composite_model_dir(model_dir):
+    if list(model_dir.glob('*/*/model.yml')):
+        raise DeserializationError(
+            '{} directory should not contain any model.yml files in any subdirectories '
+            'that are not direct children of the composite model directory'.format(model_dir))
+
+    if list(model_dir.glob('model.yml')):
+        raise DeserializationError('{} directory should not contain any model.yml files'.format(model_dir))
```

Comment:
```suggestion
    if (model_dir / 'model.yml').exists():
        raise DeserializationError('{} directory should not contain a model.yml file'.format(model_dir))
```

(There can only be one `model.yml` file in a directory.)

Reply:
Done